src/entity_builder.rs

```diff
@@ -173,6 +174,18 @@ impl EntityBuilderClone {
         self
     }
 
+    /// Add all components in `bundle` to the entity.
+    ///
+    /// If the bundle contains any component which matches the type of a component
+    /// already in the `EntityBuilder`, the newly added component from the bundle
+    /// will replace the old component and the old component will be dropped.
+    pub fn add_bundle<T: Component + DynamicBundleClone>(&mut self, bundle: T) -> &mut Self {
```

Comment:
`Component` is overly strict here, and let's use `impl Trait` for consistency with `EntityBuilder`.
```suggestion
    pub fn add_bundle(&mut self, bundle: impl DynamicBundleClone) -> &mut Self {
```